src/cljs/rems/user_settings.cljs

```diff
@@ -80,10 +80,22 @@
                    (when first-time? ; to avoid an infinite loop if the settings fail to save (e.g. unsupported language in cookies)
                      [[:check-if-should-save-language!]]))})))
 
+(rf/reg-event-fx
+ :loaded-user-settings-fail
+ (fn [{:keys [db]} [_ user-settings-error]]
+   (let [logged-in? (:user (:identity db))]
+     (if-not logged-in?
+       (if-not (= 401 (:status user-settings-error))
+         {:db db
+          :dispatch (flash-message/show-default-error! :top (str "Fetch user settings"))}
+         {:db db})
+       {:db db}))))
```

Comment:
How about we simplify this code to something like this?
```
   {:db db
    :dispatch (when (not= 401 (:status user-settings-error)))
                (flash-message/show-default-error! :top (str "Fetch user settings")))}
```

But then we could also inline this below like @opqdonut suggested because there is no necessary reference to `db`.

Btw. I think the description string should be localized here.

Reply:
Yes, I agree, yet, just to note, we did not have it localized before.

Reply:
One more note. ACtually from an `fx` you don't need to return the `db` if you don't touch it.

Reply:
Is this a suggestion or a requirement?

Reply:
I'll let @opqdonut decide. I would definitely do this refactor, it's like a one minute thing to fix the code.

Reply:
I would like to clarify: are you talking about the localization or the one liner?

Reply:
Please simplify this code a bit. The repeated `:db db` and nested ifs make the logic look a lot more complicated than it is, at least to me. At least remove the `:db` keys, and also consider combining the conditions into one if.